x/authority/types/authorizations.go

```diff
@@ -0,0 +1,196 @@
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.fungible.MsgUpdateSystemContract",
+			AuthorizedPolicy: PolicyType_groupAdmin,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.observer.MsgUpdateObserver",
+			AuthorizedPolicy: PolicyType_groupAdmin,
+		},
+		// EmergencyPolicyMessageList
+		{
+			MsgUrl:           "/zetachain.zetacore.crosschain.MsgAddInboundTracker",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.crosschain.MsgAddOutboundTracker",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.crosschain.MsgRemoveOutboundTracker",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.fungible.MsgPauseZRC20",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.observer.MsgUpdateKeygen",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.observer.MsgDisableCCTX",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+		{
+			MsgUrl:           "/zetachain.zetacore.lightclient.MsgDisableHeaderVerification",
+			AuthorizedPolicy: PolicyType_groupEmergency,
+		},
+	}
+
+	return AuthorizationList{
+		Authorizations: authorizations,
+	}
+}
+
+// SetAuthorizations adds the authorization to the list. If the authorization already exists, it updates the policy.
+func (a *AuthorizationList) SetAuthorizations(authorization Authorization) {
+	for i, auth := range a.Authorizations {
+		if auth.MsgUrl == authorization.MsgUrl {
+			a.Authorizations[i].AuthorizedPolicy = authorization.AuthorizedPolicy
+			return
+		}
+	}
+	a.Authorizations = append(a.Authorizations, authorization)
+}
+
+// RemoveAuthorizations removes the authorization from the list. It does not check if the authorization exists or not.
+func (a *AuthorizationList) RemoveAuthorizations(authorization Authorization) {
+	for i, auth := range a.Authorizations {
+		if auth.MsgUrl == authorization.MsgUrl {
+			a.Authorizations = append(a.Authorizations[:i], a.Authorizations[i+1:]...)
+		}
+	}
+}
+
+// CheckAuthorizationExists checks if the authorization exists in the list.
+func (a *AuthorizationList) CheckAuthorizationExists(authorization Authorization) bool {
+	for _, auth := range a.Authorizations {
+		if auth.MsgUrl == authorization.MsgUrl {
+			return true
+		}
+	}
+	return false
+}
+
+// GetAuthorizedPolicy returns the policy for the given message url.If the message url is not found,
+// it returns an error and the first value of the enum.
+func (a *AuthorizationList) GetAuthorizedPolicy(msgURL string) (PolicyType, error) {
+	for _, auth := range a.Authorizations {
+		if auth.MsgUrl == msgURL {
+			return auth.AuthorizedPolicy, nil
+		}
+	}
+	// Returning first value of enum, can consider adding a default value of `EmptyPolicy` in the enum.
+	return PolicyType(0), ErrAuthorizationNotFound
+}
```

Comment:
Why not just using `GetAuthorizedPolicy` and return `(policyType, found boolean)`, what `CheckAuthorizationExists` used for?

Even if we want to be future-proof, I don't see a case where there would be another error that can be returned from it

Reply:
I added the tho functions just to make the functions easier to read , IMO its easier to comment on the different uses of the same logic if we have two separate functions .

Reply:
I don't see any usage of `CheckAuthorizationExists` in the code outside of tests, so this could be moved to a helper in test, but even for this `GetAuthorizedPolicy` would be better to use because we should also assert the actual policy type in the tests, not just check msg url existence

Reply:
My next PR uses it to check if AuthorizationExists before removing it. 
it looks more readable if the logic is like 
- `CheckAuthorizationExists` -> `RemoveAuthourization`
rather than 
- `GetAuthorizedPolicy` -> `RemoveAuthorization`
But I can remove it here and add it back in if needed